app/src/main/java/com/karumi/ui/domain/Result.kt

```diff
@@ -0,0 +1,25 @@
+package com.karumi.ui.domain
+
+sealed class Result<R> {
+    class Left<R>(val error: Error) : Result<R>() {
+        override fun toString(): String = "Error $error"
+    }
+
+    class Right<R>(val value: R) : Result<R>() {
+        override fun toString(): String = "Value $value"
+    }
+
+    infix fun <Rp> bind(f: (R) -> (Result<Rp>)): Result<Rp> {
+        return when (this) {
+            is Result.Left<R> -> Left(this.error)
+            is Result.Right<R> -> f(this.value)
+        }
+    }
+
+    infix fun <Rp> seq(e: Result<Rp>): Result<Rp> = e
+
+    companion object {
+        fun <R> success(value: R) = Result.Right(value)
+        fun <R> error(error: Error) = Result.Left<R>(error)
+    }
+}
```

Comment:
If you implemented this class it's great! I guess we don't need more right now. But I'd rename this class as ``Either`` and if it's possible, I'd use [this](https://github.com/kategory/kategory/blob/master/kategory-core/src/main/kotlin/kategory/data/Either.kt) or [this](https://github.com/MarioAriasC/funKTionale/blob/master/funktionale-either/src/main/kotlin/org/funktionale/either/Either.kt) implementation so we don't need to re-invent the wheel. It seems the libraries implementations are more complete 😃

Reply:
I've been using funKTionale and it fits perfectly as it is separated in tons of modules that we can load whenever we want (`funktionale-either` is what we want here)

Reply:
I took the Kotlin blog implementation :), I renamed from Either to Result because a lot of people don't know what Either means. But I will import funktionale-either.

Reply:
What is your opinion about limit the Left type to Error except open to any kind of type?

Reply:
Either represents the disjunction mathematical concept where the left side of the expression can be an error or not. I'd not force the usage of an error in the left side.

Reply:
If we are going to use always for symbolizing an error? is not good limit the inheritance?

Reply:
Most of the times we will use it with errors, but not always. In this project, we don't even need to use errors. If you want to port the original version of this kata you don't even need to use ``Either`` or ``Result``.

Reply:
I'm asking for learn :) beacuse If the first time that I implement this kind of things. I like understand why we do some things. Original kata is not writting in kotlin, I like adjust the code to kotlin for learn and discover different ways to develop. Remember that the swift Kata is not similar to the Java one ;)

Reply:
Don't read me wrong. I'm just trying to give you my feedback as you requested. If you read the [Swift](https://github.com/Karumi/KataSuperHeroesIOS/blob/8f7cbf815b8ac786bd03d68ffd6f0c02d203b158/KataSuperHeroes/GetSuperHeroes.swift#L19) implementation of this kata we don't handle errors. If you have any doubt or any concrete question I did not answer, please let me know.